docs/how-to-guides/use-completion.rst

```diff
@@ -0,0 +1,58 @@
+.. _use_completion:
+
+Use the completion module
+=========================
+
+Craft-cli provides a completion module for auto-generating bash completion scripts for
+applications using its :py:class:`Dispatcher`.
+
+In order to invoke it, the application needs to have a public function that returns
+some basic information about itself. All applications must provide their
+:py:class:`Dispatcher` and a configuration to initialise commands with. By default,
+craft-cli commands don't need to be initialised with anything, so this would be
+``None`` in the basic case.
+
+For a project named "testcraft", create the file :file:`testcraft/application.py` and
+add the following content:
+
+.. code:: python
+
+    from craft_cli import Dispatcher
+
+    def get_dispatcher() -> Dispatcher:
+        """Fill out this function to create the application's dispatcher"""
+
+    def get_app_info() -> tuple[Dispatcher, None]:
+        return get_dispatcher(), None
+
+Once the function is made, the completion module can be invoked to create the
+completion file:
+
+.. code:: shell
+
+    python3 -m craft_cli.completion testcraft testcraft.application:get_app_info > completion.sh
+
+Applications using craft-application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Commands from `craft-application`_ need their ``app_config`` dictionary in order to be
```

Comment:
Maybe an example of an app config? That could go very far in this doc.

Reply:
The `app_config` dictionary is a property generated by the `Application` object itself, and the correct way to access it is what's shown on line 56 (accessing the `app_config` property of the `Application` object).

I think a better way to clarify this would be to provide an example implementation of `get_app()` that returns the `Application`. I'll write an example below.

Reply:
https://github.com/canonical/craft-cli/pull/326#discussion_r1953384802

Reply:
Thanks! Some of the details escape me but I find your code has better signposting for users.